src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SecurityTokenElement.cs

```diff
@@ -0,0 +1,155 @@
+    /// </remarks>
+    public class SecurityTokenElement
+    {
+        private SecurityToken _securityToken;
+        private SecurityTokenHandlerCollection _securityTokenHandlers;
+        private ReadOnlyCollection<ClaimsIdentity> _subject;
+
+        /// <summary>
+        /// Creates an instance of this object using a <see cref="SecurityToken"/> object.
+        /// </summary>
+        /// <param name="securityToken">The security token this object represents.</param>
+        /// <remarks>
+        /// <see cref="GetIdentities"/> is not supported by this object if this constructor is used unless
+        /// <see cref="ValidateToken"/> is overriden.
+        /// If the securityToken passed in is a <see cref="GenericXmlSecurityToken"/> then SecurityTokenXml will 
+        /// be set to the value found in <see cref="GenericXmlSecurityToken"/>
+        /// </remarks>
+        public SecurityTokenElement(SecurityToken securityToken)
+        {
+            if (securityToken == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(securityToken));
+            }
+
+            GenericXmlSecurityToken xmlToken = securityToken as GenericXmlSecurityToken;
+            if (xmlToken != null)
+            {
+                SecurityTokenXml = xmlToken.TokenXml;
+            }
+
+            _securityToken = securityToken;
+        }
+
+        /// <summary>
+        /// Creates an instance of this object using XML representation of the security token.
+        /// </summary>
+        /// <param name="securityTokenXml">The <see cref="XmlElement"/> representation of the security token.</param>
+        /// <param name="securityTokenHandlers">The collection of <see cref="SecurityTokenHandler"/> objects that may 
+        /// be used to read and validate the security token this object represents.</param>
+        public SecurityTokenElement(XmlElement securityTokenXml, SecurityTokenHandlerCollection securityTokenHandlers)
+        {
+            if (securityTokenXml == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(securityTokenXml));
+            }
+
+            if (securityTokenHandlers == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(securityTokenHandlers));
+            }
+
+            SecurityTokenXml = securityTokenXml;
+            _securityTokenHandlers = securityTokenHandlers;
+        }
+
+        /// <summary>
+        /// Gets the XML representation of the token.
+        /// </summary>
+        /// <remarks>This property will be null unless this object was constructed using
+        /// <see cref="SecurityTokenElement(XmlElement, SecurityTokenHandlerCollection)"/>.
+        /// </remarks>
+        public XmlElement SecurityTokenXml { get; }
+        /// <summary>
+        /// Gets the security token this object represents.
```

Comment:
Needs blank line between SecurityTokenXml property and the comment block belonging to GetSecurityToken